src/app/components/tableDisplayTranslation/TableDisplayTranslation.tsx

```diff
@@ -0,0 +1,144 @@
+import { DeleteTranslation, getAllTranslation } from '@/app/Service/DictionariesServices';
+import { Box, Button, Table, TableBody, TableCell, TableContainer, TableHead, TableRow, Typography } from '@mui/material';
+import React, { useState } from 'react';
+import { useQuery, useQueryClient } from 'react-query';
+import { toast } from 'react-toastify';
+import { CiEdit } from "react-icons/ci";
+import { RiDeleteBin5Line } from "react-icons/ri";
+import DialogUpdate from '../dialogUpdateTranslation/DialogUpdate';
+
+const TableDisplayTranslation = ({ projectsId} : { projectsId: any}) => {
+  const queryClient = useQueryClient();
+  const [selectedTranslation, setSelectedTranslation] = useState<any>(null);
+
+  const [openDialog, setOpenDialog] = useState(false); 
+
+  const handleOpenDialog = (translation: any) => {
+    setSelectedTranslation(translation);
+    setOpenDialog(true);
+  };
+
+  const handleCloseDialog = () => {
+    setOpenDialog(false);
+  };
+
+    const AllTranslation = async (projectId: string) => {
+        return await getAllTranslation(projectId);
+      };
+
+      const { data: translations,} = useQuery(
+        ["allTranslation", projectsId._id],
+        () => AllTranslation(projectsId._id),
+        {
+          enabled: !!projectsId,
+          onSuccess: (data) => {
+            queryClient.invalidateQueries('allTranslation')
+          },
+          onError: (error:any) => {
+            toast.error(error.response?.data?.message || 'An error occurred');
+          },
+        }
+      );
+
+      const handelDeleteTranslation = async (projectId: string , id: string) => {
```

Comment:
use mutations instead of this regular function and also think of next , next didn't like async await  in client component